src/jit/morph.cpp

```diff
@@ -2045,12 +2045,9 @@ GenTreePtr Compiler::fgMakeTmpArgNode(
             {
                 // ToDo-ARM64: Consider using:  arg->ChangeOper(GT_LCL_FLD);
                 // as that is how FEATURE_UNIX_AMD64_STRUCT_PASSING works.
-                // We will create a GT_OBJ for the argument below.
-                // This will be passed by value in two registers.
-                assert(addrNode != nullptr);
 
-                // Create an Obj of the temp to use it as a call argument.
-                arg = gtNewObjNode(lvaGetStruct(tmpVarNum), arg);
+                // Pass by value in two registers, using a regular GT_OBJ node, created below.
+                arg->gtFlags |= GTF_DONT_CSE;
```

Comment:
I will add a comment. This is to preserve existing behavior, but there should be a TODO-1stClassStructs to remove this.

---
In reply to: [77417012](https://github.com/dotnet/coreclr/pull/7027#discussion_r77417012) [](ancestors = 77417012)